src/components/Header.tsx

```diff
@@ -0,0 +1,63 @@
+import { FormEvent, useEffect, useRef } from 'react';
+
+type Props = {
+  newTodoTitle: string;
+  setNewTodo: (value: string) => void;
+  addTodo: () => void;
+  onError: (error: string) => void;
+  onDisabled: boolean;
+};
+
+export const Header: React.FC<Props> = ({
+  newTodoTitle,
+  setNewTodo,
+  addTodo,
+  onDisabled,
+  onError,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  });
```

Comment:
This useEffect hook  lacks a dependency array, which means it will run on every render. This could lead to performance issues or unwanted behavior, such as repeatedly focusing the input on every render